### PR TITLE
drop gregor connection on desktop power events like we do on mobile

### PR DESCRIPTION
**Pull request by mmaxim**
We get bizarre problems where the internal retry loop in the RPC library seems to get stuck on resume on Windows. As a result, after coming out of sleep on Windows, there is a good chance the connection to Gregor just won't get re-made. Rather than trying to figure that out, instead hook into the power events system, and drop and reconnect the entire connection object like we do in the mobile app.

### CODE REVIEW

**Comment by mmaxim** [comment]
cc @maxtaco

File: go/service/gregor.go [hunk 1]
```diff
@@ -266,21 +266,43 @@ func (g *gregorHandler) Init() {
...
+			if !suspended {
+				monitorAction = monitorConnect
+			} else {
+				monitorAction = monitorDisconnect
+			}
+		}
+		switch monitorAction {
+		case monitorConnect:
 			// Make sure the URI is set before attempting this (possible it isn't in a race)
 			if g.uri != nil {
```
Comment (mlsteele) [inline go/service/gregor.go]: This isn't new but it's not clear to me that `g.uri` isn't a potential data race.

**Comment by mmaxim** [comment]
I'm not going to fix CI for this, some compile error in systests (not sure how that could get in there), and kbfs-server fail.

File: go/service/gregor.go (hunk 1 above)
Comment (mmaxim) [reply go/service/gregor.go]: Probably, but in practice this is just fixed to a single value.

**Pull request merged**